services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/SwapEndpointSpec.groovy

```diff
@@ -0,0 +1,887 @@
+        [flow1Src, flow1Dst, flow2Src, flow2Dst] << [
+                [flow2.source, flow1.destination, flow1.source, flow2.destination],
+                [flow1.source, flow2.destination, flow2.source, flow1.destination],
+                [flow2.destination, flow1.destination, flow2.source, flow1.source],
+                [flow1.source, flow2.source, flow1.destination, flow2.destination]
+        ]
+    }
+
+    def "Unable to swap endpoints for existing flow and non-existing flow"() {
+        given: "An active flow"
+        def switchPair = topologyHelper.getNeighboringSwitchPair()
+        def flow1 = flowHelper.randomFlow(switchPair)
+        def flow2 = flowHelper.randomFlow(switchPair)
+        flowHelper.addFlow(flow1)
+        flow2.id = NON_EXISTENT_FLOW_ID
+
+        when: "Try to swap endpoints for existing flow and non-existing flow"
+        northbound.swapFlowEndpoint(
+                new SwapFlowPayload(flow1.id, flow1.source, flow2.destination),
+                new SwapFlowPayload(flow2.id, flow2.source, flow1.destination))
+
+        then: "An error is received (404 code)"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 404
+        exc.responseBodyAsString.to(MessageError).errorMessage == "Can not swap endpoints for flows: " +
+                "Flow ${flow2.id} not found"
+
+        and: "Delete the flow"
+        flowHelper.deleteFlow(flow1.id)
+    }
+
+    @Unroll
+    def "Unable to swap #endpointsPart for two flows (#description)"() {
+        given: "Three active flows"
+        flowHelper.addFlow(flow1)
+        flowHelper.addFlow(flow2)
+        flowHelper.addFlow(flow3)
+
+        when: "Try to swap #endpointsPart for two flows"
+        northbound.swapFlowEndpoint(
+                new SwapFlowPayload(flow1.id, flow1Src, flow1Dst),
+                new SwapFlowPayload(flow2.id, flow2Src, flow2Dst))
+
+        then: "An error is received (409 code)"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 409
+        exc.responseBodyAsString.to(MessageError).errorMessage == "Can not swap endpoints for flows: " +
+                "Requested $conflictingEndpoint endpoint for flow '${flow1.id}' conflicts with existing " +
+                "$conflictingEndpoint endpoint for flow '${flow3.id}'."
+
+        and: "Delete flows"
+        [flow1, flow2, flow3].each { flowHelper.deleteFlow(it.id) }
+
+        where:
+        endpointsPart << ["ports and vlans", "ports and vlans", "vlans", "vlans", "ports", "ports", "ports", "ports"]
+        description << ["the same ports and vlans on src switch",
+                        "the same ports and vlans on dst switch",
+                        "the same vlans on the same port on src switch",
+                        "the same vlans on the same port on dst switch",
+                        "no vlan vs vlan on the same port on src switch",
+                        "no vlan vs vlan on the same port on dst switch",
```

Comment:
We should keep in mind that very soon we are going to allow novlan and vlan flows on the same switch-port
https://github.com/telstra/open-kilda/pull/2417